vscode/microsoft-kiota/package.json

```diff
@@ -532,6 +535,7 @@
     "glob": "^11.0.1",
     "mocha": "^11.1.0",
     "sinon": "^19.0.2",
```

Comment:
if we're going to use jest, should we remove sinon?

Reply:
I can remove it so that sinon remains inside the vscode environment while this uses jest. So that the final separation is easier

Reply:
> if we're going to use jest, should we remove sinon?

This has been removed.

> even across packages/products, I don't think we should have a different set of test/mock frameworks. It doubles up the costs of maintenance, ramping up people, etc...

You're right. jest was appropriate for this module because 
1. it has an easy to set up and get started process. 
2. vscode extensions exist to make the detection and running of the tests easier which I didn't find for mocha.
3. I needed to separate tests that require a vscode emulator and those that don't for faster testing

It will be worth investing in changing the rest of the tests in the vscode environment to jest (if possible) at a time where the deadline is not as tight

Reply:
Thank you for the additional information.
We're only talking about [3 unit tests](https://github.com/search?q=repo%3Amicrosoft%2Fkiota+sinon+language%3ATypeScript&type=code&l=TypeScript) and [one setup file](https://github.com/microsoft/kiota/blob/c5d47a0784be7bf10fb3a45e9a07974ad2f285e0/vscode/microsoft-kiota/src/test/suite/index.ts#L4)
Can I ask that you give it a try? it should be fairly quick and will avoid planning overhead as well as potential confusion.

Reply:
I did.. it took up most of my day trying to redo the setup. I think I can give it a shot again without the looming deadlines

Reply:
I've made some time. Changes are coming in

Reply:
This is done

Reply:
Thank you for making the changes!

Do you think sinon could also be removed?